docs/mvc/tutorials/mvc-with-entity-framework/sample/src/ContosoBooks/Models/SampleData.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Microsoft.Data.Entity.SqlServer;
+using System;
+using System.Threading.Tasks;
+using Microsoft.Framework.DependencyInjection;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Data.Entity;
+
+namespace ContosoBooks.Models
+{
+    public static class SampleData
+    {
+        public static void Initialize(IServiceProvider serviceProvider)
+        {
+            using (var db = serviceProvider.GetService<BookContext>())
+            {
+                var sqlServerDb = db.Database as SqlServerDatabase;
```

Comment:
You shouldn't need all this casting, DbContext.Database.EnsureCreated() and EnsureDeleted() are general abstractions that should work for what you want.